code/app/src/main/java/com/example/the_tarlords/data/QR/QRScanActivity.java

```diff
@@ -91,6 +92,10 @@ public void QrtoEvent(String QrID) {
                                 User user = new User();
                                 user.setUserId(userId);
                                 event.setCheckIn(user, true);
+                                // Seperate from check-In so its fine here
+                                ShareLocation shareLocationDialog = new ShareLocation(event.getId(),event.getName());
```

Comment:
The reason I had it in check in was more to make it so that it would only request permission on a successful check-in, but I'm thinking of making an attendance callback interface so maybe I could work this into that